src/metrics-gatherer.ts

```diff
@@ -134,81 +183,135 @@ export class MetricsGatherer {
 		labels: LabelSet = {},
 		custom: CustomParams = {},
 	) {
-		// if exists, bail early
-		if (this.existMap[name]) {
-			return;
-		}
-		// if already exists with another kind, throw error
-		if (this.kinds[name] && this.kinds[name] !== kind) {
-			throw new MetricsGathererError(
-				`tried to use ${name} twice - first as ` +
-					`${this.kinds[name]}, then as ${kind}`,
-			);
-		}
-		// if not described, describe (poorly)
-		if (!(name in this.descriptions)) {
-			this.descriptions[name] = `undescribed ${kind} metric`;
+		try {
+			// if exists, bail early
+			if (this.existMap[name]) {
+				return;
+			}
+			// if already exists with another kind, throw error
+			if (this.kinds[name] && this.kinds[name] !== kind) {
+				this.err(
+					new MetricsGathererError(
```

Comment:
Might be worth just throwing this and handling it with everything else in the `catch` below